tests/test_cli/test_gui.py

```diff
@@ -16,56 +16,59 @@
 #   limitations under the License.
 #
 # ------------------------------------------------------------------------------
-
 """This test module contains the tests for the `aea gui` sub-command."""
-
 import json
 import os
-import subprocess  # nosec
 import sys
 import tempfile
-import time
 from pathlib import Path
+from unittest.case import TestCase
 
 import jsonschema
 from jsonschema import Draft4Validator
 
+import pytest
+
+from tests.common.pexpect_popen import PexpectSpawn
+
 from ..conftest import (
     AGENT_CONFIGURATION_SCHEMA,
-    CLI_LOG_OPTION,
     CONFIGURATION_SCHEMA_DIR,
     tcpping,
 )
 
 
-class TestGui:
+class TestGui(TestCase):
     """Test that the command 'aea gui' works as expected."""
 
-    @classmethod
-    def setup_class(cls):
+    def setUp(self):
         """Set the test up."""
-        cls.schema = json.load(open(AGENT_CONFIGURATION_SCHEMA))
-        cls.resolver = jsonschema.RefResolver(
-            "file://{}/".format(Path(CONFIGURATION_SCHEMA_DIR).absolute()), cls.schema
+        self.schema = json.load(open(AGENT_CONFIGURATION_SCHEMA))
+        self.resolver = jsonschema.RefResolver(
+            "file://{}/".format(Path(CONFIGURATION_SCHEMA_DIR).absolute()), self.schema
         )
-        cls.validator = Draft4Validator(cls.schema, resolver=cls.resolver)
+        self.validator = Draft4Validator(self.schema, resolver=self.resolver)
 
-        cls.agent_name = "myagent"
-        cls.cwd = os.getcwd()
-        cls.t = tempfile.mkdtemp()
-        os.chdir(cls.t)
-        cls.proc = subprocess.Popen(  # nosec
-            [sys.executable, "-m", "aea.cli", *CLI_LOG_OPTION, "gui"]
-        )
-        time.sleep(10.0)
+        self.agent_name = "myagent"
+        self.cwd = os.getcwd()
+        self.t = tempfile.mkdtemp()
+        os.chdir(self.t)
 
+    @pytest.mark.integration
     def test_gui(self):
         """Test that the gui process has been spawned correctly."""
-        assert tcpping("127.0.0.1", 8080)
+        self.proc = PexpectSpawn(  # nosec
+            sys.executable,
+            ["-m", "aea.cli", "-v", "DEBUG", "gui"],
+            encoding="utf-8",
+            logfile=sys.stdout,
+        )
+        self.proc.expect_exact(["Running on http://"], timeout=20)
+
+        assert tcpping("localhost", 8080)
```

Comment:
Note, this must be "127.0.0.1" instead of "localhost"..more reliable

Reply:
done